### PR TITLE
dex: use spec defined epoch duration if provided value is not valid

### LINKED ISSUE #300: dcrdex config should set a default epoch duration
Fair enough, will check for 0 there.  Also seems reasonable to replace 0/-1/omitted with a the specd default.

_Originally posted by @chappjc in https://github.com/decred/dcrdex/pull/283_

### PR DESCRIPTION
**Pull request by ukane-philemon**
This diff ensures a market's epoch duration is set to the [spec-defined epoch duration](https://github.com/decred/dcrdex/blob/master/spec/admin.mediawiki#exchange-variables) if the config epoch duration is invalid. Closes #300.

### CODE REVIEW

File: dex/market.go [hunk 1]
```diff
@@ -53,6 +59,12 @@ func NewMarketInfo(base, quote uint32, lotSize, rateStep, epochDuration uint64,
 	if err != nil {
 		return nil, err
 	}
+
+	// Check for sensible epoch duration.
+	if epochDuration <= 0 {
```
Comment (JoeGruffins) [inline dex/market.go:64]: nit: uint64 cannot be < 0

**Review by JoeGruffins — APPROVED** [review]

File: dex/market.go (hunk 1 above)
Comment (ukane-philemon) [reply dex/market.go:64]: > nit: uint64 cannot be < 0

You mean to change the comment, right?

Comment (chappjc) [reply dex/market.go:64]: He means you only need to check for `== 0`.

File: dex/market.go [hunk 2]
```diff
@@ -9,6 +9,12 @@ import (
 	"strings"
 )
 
+const (
+	// defaultEpochDuration is the spec defined default markets epoch duration
+	// in milliseconds.
+	defaultEpochDuration uint64 = 60000
```
Comment (chappjc) [inline dex/market.go:15]: Turns out that 60 sec is an extremely long epoch.  I think we should revise the spec to make it 10 sec, 20 tops.

Comment (ukane-philemon) [reply dex/market.go:15]: > I think we should revise the spec to make it 10 sec, 20 tops.

20 sec should be good.

**Review by chappjc — APPROVED** [review]

**Pull request merged**